includes/wikia/services/CommunityDataService.class.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+class CommunityDataService extends WikiaService {
+	const CURATED_CONTENT_VAR_NAME = 'wgWikiaCuratedContent';
+	private $curatedContentData = [ ];
+
+	private $cityId;
+
+	function __construct( $cityId ) {
+		parent::__construct();
+		$this->cityId = $cityId;
+	}
+
+	public function setCuratedContent( $data ) {
+		$ready = $this->isLegacyFormat( $data ) ? $this->toNew( $data ) : $data;
+		$status = WikiFactory::setVarByName( self::CURATED_CONTENT_VAR_NAME, $this->cityId, $ready );
+		if ( $status ) {
+			wfWaitForSlaves();
+			$this->curatedContentData = $ready;
+		}
+		return $status;
+	}
+
+	public function getCuratedContent() {
+		return $this->curatedContentData();
+	}
+
+	public function hasCuratedContent() {
+		return !empty( $this->getCuratedContent() );
+	}
+
+	/**
+	 * Returns both curated and optional sections merged together
+	 * @return array
+	 */
+	public function getNonFeaturedSections() {
+		$curated = $this->getCurated();
+		$optional = $this->getOptional();
+		if ( !empty( $optional ) ) {
+			$curated[] = $optional;
```

Comment:
Are you sure about this type of assignments? It looks like getCurated returns an array of items, getOptional a single item, and it may be confusing especially because of 'merge' used in description. If there's only one item in optional, I'd suggest making it explicit; alternatively, unify the format so both are an array.